osu.Game/Scoring/ScoreManager.cs

```diff
@@ -132,9 +132,10 @@ public void GetTotalScore([NotNull] ScoreInfo score, [NotNull] Action<long> call
         /// <returns>The total score.</returns>
         public async Task<long> GetTotalScoreAsync([NotNull] ScoreInfo score, ScoringMode mode = ScoringMode.Standardised, CancellationToken cancellationToken = default)
         {
-            // TODO: ??
-            // if (score.Beatmap == null)
-            //     return score.TotalScore;
+            // TODO: This is required for playlist aggregate scores. They should likely not be getting here in the first place.
+            // ReSharper disable once ConditionIsAlwaysTrueOrFalse
+            if (score.BeatmapInfo == null)
+                return score.TotalScore;
```

Comment:
Of course it has 😵‍💫

Reply:
I've pushed a fix for this for now. Was looking to add test coverage but ran into more (slightly unrelated) issues in the process, so that may come later.

Reply:
Test coverage added via #16489.